Fix issue #80 and almost complete

bagustris opened the pull request:
- Fix issue #80 
- Drop lesson 5 (directory structure) since the content is almost empty and the basics were already covered in shell-novice
- Fix many typos (cut -> cat, etc)
- Fix permission issue (#78)
- Add shebang (#31, #10) 
- Add files for exercise (awk) 
- Add code example (shebang)

Preview available at: https://bagustris.github.io/shell-extras/

bagustris committed:
Create config.md

bagustris committed:
Create README.md

bagustris committed:
Add files via upload

bagustris committed:
add help in _extras

bagustris committed:
update awk, remove dir, etc.

bagustris committed:
update ch 9 and 10

bagustris committed:
update ch 9 and 10 again to show overview

bagustris committed:
rename md files due to remove ch dir structure

bagustris committed:
rm lesson 5

bagustris committed:
update lesson titles

bagustris committed:
fix typo in lesson 8 -> excercise

Commit:
update lessons 8-9

bagustris committed:
Update 09-awk.md

kynan commented on _episodes/08-command-substitution.md:
```diff
@@ -170,10 +169,10 @@ whitespace to separate command, options (flags) and parameters /
 arguments.  For the same reason it is essential that the command (or
```
This section should probably be a callout box rather than a dedicated subsection? GH won't let me comment on the `### **Important**` line above though :(

bagustris replied:
Done.

kynan commented on _episodes/08-command-substitution.md:
```diff
@@ -237,14 +235,13 @@ loops.
 <!-- solution: for file in *.pdb; do sort $file > $(basename $file .pdb).sorted; done -->
+{: .challenge}
 
 ## Closing remarks
```
This chapter is missing "Key Points", so the respective box is empty :)

bagustris replied:
Done

kynan commented on _episodes/09-awk.md:
This chapter is also missing "Key Points"

bagustris replied:
Done

kynan commented on _extras/config.md:
Where do pages under `_extras` end up in the rendered output?

bagustris replied:
I actually added `help.md` and `config.md` for backup when these references disappear, I didn't touch other setting.

bagustris committed:
fix missing keypoints

bagustris committed:
add data for exercise

bagustris committed:
update permissions, change all to others

bagustris committed:
add example file for shebang

bagustris committed:
update shebang

openhands-agent committed:
Address reviewer feedback: fix keypoints formatting and add YAML front matter to extras

- Fix grammar error in episode 8 keypoints (can achieved -> can be achieved)
- Add quotes to keypoints in episodes 8 and 9 for proper YAML formatting
- Add YAML front matter to _extras/config.md and _extras/help.md for proper Jekyll rendering

Co-authored-by: openhands <openhands@all-hands.dev>
